plugins/sidebar/overlays/download.js

```diff
@@ -42,6 +44,46 @@ class DownloadOverlay extends React.Component {
     }
   }
 
+  download(device){
+    const { toast, workspace, logger, overlay } = this.props.irken;
+
+    const { irken, handleError, handleSuccess } = this.props;
```

Comment:
do these 2 in the inverse order and you don't have to access `this.props.irken` on line 48